### PR TITLE
BreadcrumbsBar Typescript migration

### PR DESCRIPTION
**Pull request by SergeyRoyt**
https://monday.monday.com/boards/2861766393/pulses/3389358123

### CODE REVIEW

File: src/components/BreadcrumbsBar/BreadcrumbItem/BreadcrumbContent/BreadcrumbContent.tsx [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+const ENTER_KEY = "Enter";
+const SPACE_KEY = " ";
+
+interface BreadcrumbContentProps extends VibeComponentProps {
+  isClickable?: boolean;
+  link?: string;
+  onClick?: () => void;
+  text?: string;
+  icon?: string | React.FC<iconSubComponentProps> | null;
```
Comment (hadasfa) [inline src/components/BreadcrumbsBar/BreadcrumbItem/BreadcrumbContent/BreadcrumbContent.tsx:15]: Let's convert to IconType?

File: src/components/BreadcrumbsBar/BreadcrumbItem/BreadcrumbItem.tsx [hunk 2]
```diff
@@ -1,29 +1,49 @@
...
+  /** Should item be clickable - this should be recieved from the breadcrumbsBar ( Navigation/Indication bar ). */
+  isClickable?: boolean;
+  /** If the item is clickable and the type of navigation is a link, this is the link */
+  link?: string;
+  /** If the item is clickable and the type of navigation is a function, this is the function */
+  onClick?: () => void;
+  /** Should be the current Item - mainly effects the item`s style. */
+  isCurrent?: boolean;
+  /** An Icon - If no icon needed then should be left empty. */
+  icon?: string | React.FunctionComponent<iconSubComponentProps> | null;
```
Comment (hadasfa) [inline src/components/BreadcrumbsBar/BreadcrumbItem/BreadcrumbItem.tsx:30]: Same IconType note

File: src/components/ProgressBars/LinearProgressBar/__stories__/linearProgressBar.stories.mdx [hunk 3]
```diff
@@ -1,17 +1,19 @@
...
-import { SPINNER, STEPPER, SKELETON } from "../../../../storybook/components/related-components/component-description-map";
+import {
+  SKELETON,
+  SPINNER,
+  STEPPER
+} from "../../../../storybook/components/related-components/component-description-map";
 import { Link } from "../../../../storybook/components";
-import BreadcrumbsBar from "../../../BreadcrumbsBar/BreadcrumbsBar.jsx";
-import BreadcrumbItem from "../../../BreadcrumbsBar/BreadcrumbItem/BreadcrumbItem.jsx";
+import BreadcrumbsBar from "../../../BreadcrumbsBar/BreadcrumbsBar.tsx";
```
Comment (hadasfa) [inline src/components/ProgressBars/LinearProgressBar/__stories__/linearProgressBar.stories.mdx:15]: I think the extension is not really needed I think (.tsx)

**Review by hadasfa — APPROVED** [review]
Approved with small notes :)

File: src/components/BreadcrumbsBar/BreadcrumbItem/BreadcrumbContent/BreadcrumbContent.tsx (hunk 1 above)
Comment (SergeyRoyt) [reply src/components/BreadcrumbsBar/BreadcrumbItem/BreadcrumbContent/BreadcrumbContent.tsx:15]: I think we should fix IconType. Current implementation results in these errors
image
Maybe in this case it should be like this `keyof JSX.IntrinsicElements`?

**Pull request merged**